CODEOWNERS

```diff
@@ -1,6 +1,7 @@
 /elements/pl-hidden-hints @eliotwrobson
 /elements/pl-order-blocks @SethPoulsen
 /elements/pl-rich-text-editor @jonatanschroeder
+/elements/pl-dataframe @coatless @eliotwrobson
```

Comment:
Should we name this `pl-pandas-dataframe` instead? This currently seems to be very closely tied to Pandas, so I'd like to reflect this in the name if this isn't generalizable to _other_ types of dataframes (whatever those may be).

Reply:
The goal here would be to move away from being bound to just Python's `pandas` and focus on how best to display tabular data for any language. For example, in the current languages supported in `pl-variable-output`, we have `DataFrame` equivalents of:

- R's [`dataframe()`](https://stat.ethz.ch/R-manual/R-devel/library/base/html/data.frame.html) or [`tibble()`](https://tibble.tidyverse.org/)
- MATLAB's community-led [dataframe approach](https://www.mathworks.com/matlabcentral/fileexchange/95258-dataframe-timeframe)
- Mathematica's [`Dataset`](https://reference.wolfram.com/language/ref/Dataset.html)

If Julia was added, then we would have: 

- Julia's [`DataFrame.jl`](https://dataframes.juliadata.org/stable/)

Reply:
`pl-variable-output` displays data _in_ different formats, but AFAICT this element is about displaying data _from_ a specific format. That feels like an important distinction. Folks can't write questions in MATLAB or Mathematica or Julia, so how would they get a `DataFrame.jl` to pass to this element in the first place?

Reply:
@nwalters512 I think `pl-dataframe` is similar to `pl-variable-output`'s goal of displaying data in different formats. Not necessarily a _specific_ format, e.g. `pandas`. This element has the same limitation as `pl-variable-output`, we need to use _Python_  and its libraries to generate the underlying object. 

For instance, with the incorporation of https://github.com/PrairieLearn/PrairieLearn/issues/6787, we now have the ability to highlight _R_ data types and indices. We can further add re-construction code in _R_ to re-build the data frame or provide a file download option using a language agnostic file format (e.g. [`arrow`](https://arrow.apache.org/docs/python/pandas.html)) containing the data frame.

If you have a few seconds to address external dependency, we can even allow for the _visual_ table to gain interactive sorting features described in https://github.com/PrairieLearn/PrairieLearn/issues/6788 (@eliotwrobson and myself are not sure how to resolve the external dependency).

So, I view this element as being a bit more than just a _pandas_ specific implementation, but more so first-class support for displaying and using tabular data. We're just slowly moving to incorporate other export formats.

Reply:
I also don't think it's unreasonable to think we could support different types of Python dataframes in the future, for example, [Polars](https://www.pola.rs/) is becoming more popular.